src/main/java/seedu/moneygowhere/storage/LocalStorage.java

```diff
@@ -197,18 +198,35 @@ private Expense createExpense(Node node) throws LocalStorageLoadDataInputError {
         Element element = (Element) node;
         String name = element.getElementsByTagName(XML_EXPENSE_NAME_ELEMENT)
                 .item(0).getTextContent();
-        LocalDateTime dateTime = LocalDateTime.parse(element
-                .getElementsByTagName(XML_EXPENSE_DATETIME_ELEMENT).item(0).getTextContent());
+        assert !(name.isEmpty() || name.trim().isEmpty()) : "There must be a name";
+
         String description = element.getElementsByTagName(XML_EXPENSE_DESCRIPTION_ELEMENT)
                 .item(0).getTextContent();
-        NodeList amountNodeList = element.getElementsByTagName(XML_EXPENSE_AMOUNT_ELEMENT);
-        BigDecimal amount = new BigDecimal(amountNodeList.item(0).getTextContent());
-        String currency = amountNodeList.item(0).getAttributes()
-                .getNamedItem(XML_EXPENSE_AMOUNT_CURRENCY_ATTRIBUTE).getTextContent();
+        if (description.isEmpty() || description.trim().isEmpty()) {
+            description = null;
+        }
+
         String category = element.getElementsByTagName(XML_EXPENSE_CATEGORY_ELEMENT)
                 .item(0).getTextContent();
+        if (category.isEmpty() || category.trim().isEmpty()) {
+            category = null;
+        }
+
         String remark = element.getElementsByTagName(XML_EXPENSE_REMARKS_ELEMENT)
                 .item(0).getTextContent();
+        if (remark.isEmpty() || remark.trim().isEmpty()) {
+            remark = null;
+        }
+
+        NodeList amountNodeList = element.getElementsByTagName(XML_EXPENSE_AMOUNT_ELEMENT);
+        String currency = amountNodeList.item(0).getAttributes()
+                .getNamedItem(XML_EXPENSE_AMOUNT_CURRENCY_ATTRIBUTE).getTextContent();
+        assert !(currency.isEmpty() || currency.trim().isEmpty()) : "There must be a currency";
```

Comment:
May I ask is there a reason why assert is used to perform input validation? I don't believe assert is enabled by default in most cases.

Reply:
Sorry this was a mistake on my part. Will move assertions into constructor of the objects and use exception in the loading of data.